cloud/amazon/iam_cert.py

```diff
@@ -273,10 +273,18 @@ def main():
                                                     iam.get_all_server_certs().\
                                                     list_server_certificates_result.\
                                                     server_certificate_metadata_list]
-    orig_bodies = [iam.get_server_certificate(thing).\
-                  get_server_certificate_result.\
-                  certificate_body \
-                  for thing in orig_certs]
+    orig_bodies = []
+    for thing in orig_certs:
+        try:
+            orig_bodies.append(iam.get_server_certificate(thing).\
+                               get_server_certificate_result.\
+                               certificate_body)
+        except boto.exception.BotoServerError, err:
+            if err.status == 404 and err.code == 'NoSuchEntity':
+                module.exit_json(msg="Warning: %s" % err.message)
```

Comment:
this will exit the module, when i mentioned 'warning's i meant it as part of a normal return if you did not want to exit at this point. Also it is it's own key in 2.0 `exit_json(warnings=warnings, msg=msg, ...)`